### ISSUE TITLE
glint-expect-error is failing, I think because an import's type is lost (becomes 'any'?)

### ISSUE THREAD

**Issue by NullVoxPopuli** [typed-ember/glint#921]
<img width="516" height="202" alt="Image" src="https://github.com/user-attachments/assets/c6f6c9b4-fe32-4867-9906-478ec0f65a72" />

Repro:
```
git clone git@github.com:universal-ember/ember-primitives.git
cd ember-primitives
git checkout load-component
git checkout 14f1db0
pnpm install
cd test-app
code .
```

The file is: `tests/on-resize/rendering-test.gts` and the issue is towards the bottom on line 234

This is using pre-release glint extension in vscode and 2.0.0-alpha.3 of the glint main deps and 1.6.0-alpha.2 of `@glint/template`

Using Volar Labs, I see the transform becomes:
```ts

      await render(
        ({} as typeof import("@glint/environment-ember-template-imports/-private/dsl")).templateExpression(function(__glintRef__, __glintDSL__: typeof import("@glint/environment-ember-template-imports/-private/dsl")) {
{
const __glintY__ = __glintDSL__.emitElement("div");
__glintDSL__.applyAttributes(__glintY__.element, {
"data-test": "",


});
__glintDSL__.applyModifier(__glintDSL__.resolve(onResize)(__glintY__.element, ));
}
__glintRef__; __glintDSL__;
// begin directive placeholders
// @ts-expect-error expect-error
;
// end directive placeholders
})
      );
    });
```

which kinda makes sense as to why the `ts-expect-error` is reporting as "unused" because the modifier is no where near there?

I copied the Volar Labs output to a ts file (which I had to save before some of the imports would resolve),

Here is what things look like here:

<img width="1224" height="640" alt="Image" src="https://github.com/user-attachments/assets/9d8e47e5-b597-4d69-aa65-1b81ec686502" />

so, this is very similar, if not the same as #920

**Comment by NullVoxPopuli** [comment]
Even moving the glint-expect-error in to attribute space doesn't help

<img width="1663" height="963" alt="Image" src="https://github.com/user-attachments/assets/f2122932-0865-4ecc-b3d9-47d713039723" />

**Comment by NullVoxPopuli** [comment]
Note tho that I don't have this problem in neovim: 

<img width="1846" height="1554" alt="Image" src="https://github.com/user-attachments/assets/5b4d7457-8dd5-4b15-9ed9-92197974f06f" />

so maybe it was specific to the glint extension? 

or maybe the __glintDSL__ variables being any are just _too_ forgiving in general

**Comment by machty** [comment]
I will take a look at this and repro shortly but the "bottom-hoisted" ts-expect-error scheme is intentional. There is a very delicate set of constraints for how to push the glint-expect-error functionality into the VirtualCode implementation so that it works in all contexts (tsplugin + tsc, etc). The most recent change required me hoisting the ts-expect-errors at the bottom because it was the only way i could guarantee that the ts-expect-errors were emitted at a syntactic top-level / statement-level (previous attempts would occasional insert a `// @ts-expect-error` somewhere in the middle of an expression which broke the transformed code).

The reason/way this works in the general case (though obviously there are some issues you're seeing) is that we use the `shouldReport` callback on the `verification` mapping to track if errors occur around the code at the TOP and then we use it to control whether the placeholder bottom-hoisted `ts-expect-error` suppresses or lets through the "unused ts-expect-error" diagnostic at the bottom. With this scheme, the placeholder ts-expect-error doesn't need to appear next to the line of code covered by the diagnostic; rather it can be "hoisted" at the bottom like we do.
